src/runtime_src/core/tools/common/Report.h

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef __Report_h_
+#define __Report_h_
+
+// Please keep eternal include file dependencies to a minimum
+#include "core/common/device.h"
+#include <boost/property_tree/ptree.hpp>
+#include <iostream>
+#include <string>
+#include <vector>
+
+class Report {
+ public:
+  // Supported JSON schemas.
+  // 
+  // Remember to update the initialization of Report::m_schemaVersionMapping 
+  // if new enumeration values are added
+  enum class SchemaVersion  {
+    TEXT,
+    UNKNOWN,
+    JSON_INTERNAL,
+    JSON_20201,

```

Comment:
Use lower case enumeration variables for enum classes.
Otherwise something like TEXT and UNKNOWN have a big risk of being substituted by preprocessing.

Reply:
Made changes as suggested.